docs/porting.md

```diff
@@ -406,7 +469,74 @@ int function_accessing_the_cache(void) {
     [this NumPy PR](https://github.com/numpy/numpy/pull/27011) that ports all
     `PyThread_type_lock` usages to `PyMutex`.
 
-### Dealing with thread-unsafe libraries
+## Fixing thread-unsafe tests.
+
+Many existing tests are written using global state. This is not a problem if the
+test only runs once, but if you would like to use your tests to check for
+possible thread safety issues by running existing tests on many threads, you
+will likely need to update the tests to eliminate use of global state.
+
+Since tests using global state are inherently racey, this means that test
+failures associated with these tests are also inherently flakey. If you see
+tests failing intermittently, you should not discount that you are using global
+state in a test, or even inadvertantly using global state in PyTest itself.
+
+#### PyTest is not thread-safe
+
+See [the PyTest
+docs](https://docs.pytest.org/en/stable/explanation/flaky.html#thread-safety)
+for more information about this. While tests can manage their own threads, you
+should not assume that functionality provided by PyTest is thread-safe.
+
+Functionality that is known not to be thread-safe includes:
+
+- [`pytest.warns`](https://docs.pytest.org/en/stable/reference/reference.html#pytest.warns),
+    it relies on `warnings.catch_warnings`, which is not thread safe.
+- The [`tmp_path`](https://docs.pytest.org/en/stable/reference/reference.html#std-fixture-tmp_path) and [`tmpdir`](https://docs.pytest.org/en/stable/reference/reference.html#std-fixture-tmpdir) fixtures, since they rely on the filesystem
```

Comment:
```suggestion
- The [`tmp_path`](https://docs.pytest.org/en/stable/reference/reference.html#std-fixture-tmp_path)
  and [`tmpdir`](https://docs.pytest.org/en/stable/reference/reference.html#std-fixture-tmpdir)
  fixtures, since they rely on the filesystem.
```